nsis/src/nsis/NsisLoader.java

```diff
@@ -144,26 +180,46 @@ public Data createData(Program program, Listing listing, Address address,
 		return null;
 	}
 
-	private void processBlockHeaders(Program program, TaskMonitor monitor,
-			BinaryReader reader, long nsis_header_offset) {
-		int block_header_offset = NsisScriptHeader.getHeaderSize();
+	/**
+	 * Initializes the block headers and adds them to the "Program Trees" view in
+	 * Ghidra.
+	 * 
+	 * @param program
+	 * @param reader
+	 * @param startingAddr, the Address where the nsis script header starts
+	 * @throws IOException
+	 * @throws AddressOverflowException
+	 * @throws MemoryConflictException
+	 * @throws DuplicateNameException
+	 * @throws LockException
+	 */
+	private void initBlockHeaders(Program program, BinaryReader reader, Address startingAddr,
+			FileBytes fileBytes) throws IOException, LockException, DuplicateNameException,
+			MemoryConflictException, AddressOverflowException {
+		int block_header_offset = 0;
+
+		Memory memory = program.getMemory();
+		MemoryBlock new_block = memory.createInitializedBlock(".block_headers", startingAddr,
```

Comment:
Does the data contain just block headers or all of the blocks (header and data)?

Also, since we later loop through the list of blocks, do we want to call `memory.createInitializedBlock` for each block and give it a unique name?

Reply:
For now it contains all of the blocks since I didn't research what comes after the block headers. I definitely think we should add memory blocks for the other blocks and give them names. I'm thinking of opening  an issue for that and create a PR later on to address that specifically. What do you think?

Reply:
Sounds good.